server/migrations/20200416160146-reformat-notification-data.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable quotes */
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    const notifications = await queryInterface.sequelize.query(`SELECT * FROM "Notifications";`);
+    await Promise.all(notifications[0].map(async (notif) => {
+      let data = JSON.parse(notif.notification_data);
+      if (data.thread_id) {
+        data['root_id'] = Number(data.thread_id);
+        data['root_type'] = 'discussion';
+        data['root_title'] = data.thread_title;
+        delete data.thread_id;
+        delete data.thread_title;
+      } else if (data.comment_id) {
+        const root = data.object_id || data.root_id;
+        const [prefix, id] = root.split("_");
```

Comment:
This must be the offending like per @raykyri's comment

Reply:
The migration is working for me now, using the latest production dump, so I guess this is resolved.